tasks/install_powershell.ps1

```diff
@@ -0,0 +1,50 @@
+[CmdletBinding()]
+Param(
+	[String]$version
+)
+# If an error is encountered, the script will stop instead of the default of "Continue"
+$ErrorActionPreference = "Stop"
+
+$arch = if ((Get-WmiObject Win32_OperatingSystem).OSArchitecture -match '^32') {"x86"} else {"x64"}
+
+if ($version) {
+    $msi_source = "http://downloads.puppetlabs.com/windows/puppet5/puppet-agent-${version}-${arch}.msi"
```

Comment:
Hardcoded to `puppet5`. This should be parameterised

Reply:
Yes, eventually it should be.

Reply:
I think we could add that as a `collection` parameter that takes `Enum[PC1, puppet5, puppet6]`. This will increase our testing matrix a bit.

Reply:
If we had pester tests, we could easily test at the unit level and avoid acceptance tests.

Reply:
@MikaelSmith Would you like the `collection` metaparameter added in this ticket? `puppet5` is currently hard coded for the OSX support in the `install_shell.sh` implementation.

Reply:
Let's ticket as an extra subtask. Also, it's just a parameter, not a metaparameter.